### PR TITLE
QR Payment Points & Savings Features

### PR DESCRIPTION
**Pull request by Hugo0**
## Summary of Changes

- **QR Payment Points**: Added "You've earned X points!" display after Manteca and SimpleFi QR payments complete, with confetti animation
- **QR Payment Savings**: Added "If you had paid with card, it'd have cost you ~X more!" message for Argentina Manteca QR payments only (5% savings, shows cents <$1, dollars >=$1)
- **Manteca Withdraw Points**: Added points display to Manteca withdraw success view
- **Utility Functions**: Created `qr-payment.utils.ts` with savings calculation and Argentina detection logic
- **Performance**: Points fetched early via `useQuery` with 5min cache to avoid latency penalty
- **Home Page UI**: Added Points link in header (icon + text + chevron), Profile menu includes Points item with star icon, UserHeader button uses `shadowSize="3"`

## Risks

- **Low risk**: All changes additive, backward compatible
- **Backend dependency**: Requires points API to support `MANTECA_QR_PAYMENT` action type (already in use)
- **Conditional rendering**: Savings message only shows for Argentina Manteca QR payments (MERCADO_PAGO/ARGENTINA_QR3), not Brazil PIX or SimpleFi

## QA Guidelines

- **QR Payment Manteca (Argentina)**: Verify points display + savings message appear after payment
- **QR Payment Manteca (Brazil/PIX)**: Verify points display appears, savings message does NOT
- **QR Payment SimpleFi**: Verify points display appears + confetti triggers
- **Manteca Withdraw**: Verify points display appears in success view
- **Edge cases**: Test with $0.01 (shows "~1 cent"), $1.00 (shows "~$1.00"), $10.00 (shows "~$10.00")
- **Perk flow**: Verify points show even when perk is claimed (separate rewards)
- **Home page**: Verify Points link in header navigates to /points, Profile menu shows Points item

### CODE REVIEW

**Comment by Hugo0** [comment]
@coderabbitai review in depth. See .cursorrules for guidelines.

File: src/hooks/useHomeCarouselCTAs.tsx [hunk 1]
```diff
@@ -52,8 +51,8 @@ export const useBanners = () => {
         }
 
         if (!isUserKycApproved && !isUserBridgeKycUnderReview) {
-            _banners.push({
-                id: 'kyc-banner',
+            _carouselCTAs.push({
```
Comment (kushagrasarathe) [inline src/hooks/useHomeCarouselCTAs.tsx:54]: thank you hugo 🙏

File: src/app/(mobile-ui)/withdraw/manteca/page.tsx [hunk 2]
```diff
@@ -323,6 +344,18 @@ export default function MantecaWithdrawFlow() {
                             <h1 className="text-sm font-normal text-grey-1">to {destinationAddress}</h1>
                         </div>
                     </Card>
+
+                    {/* Points Display */}
+                    {pointsData?.estimatedPoints && (
+                        <div ref={pointsDivRef} className="flex justify-center gap-2">
```
Comment (kushagrasarathe) [inline src/app/(mobile-ui)/withdraw/manteca/page.tsx:341]: qn non blocking: why is ref neded?

File: src/app/(mobile-ui)/qr-pay/page.tsx [hunk 3]
```diff
@@ -313,6 +318,21 @@ export default function QRPayPage() {
         }
     }, [paymentProcessor, simpleFiPayment, paymentLock?.code, paymentLock?.paymentAgainstAmount, amount])
 
+    // Fetch points early to avoid latency penalty - fetch as soon as we have usdAmount
+    // This way points are cached by the time success view shows
+    // Only Manteca QR payments give points (SimpleFi does not)
+    const { data: pointsData } = useQuery({
```
Comment (kushagrasarathe) [inline src/app/(mobile-ui)/qr-pay/page.tsx]: could have been a hook, used two times

**Review by kushagrasarathe — APPROVED** [review]
one suggestion on abstraction of points fetching

**Comment by Hugo0** [comment]
@coderabbitai review new changes in depth and evaluate if PR is ready to merge

**Comment by Hugo0** [comment]
@coderabbitai review in depth again.
